src/parser.rs

```diff
@@ -0,0 +1,219 @@
+use crate::ast::{Identifier, PluginBuilder, PluginOption, Tag, Task, TaskBuilder, Value};
+use pest::iterators::Pairs;
+use pest_derive::Parser;
+
+#[derive(Parser)]
+#[grammar = "dsl.pest"]
+struct DSLParser;
+
+fn create_task(pairs: Pairs<Rule>, task_builder: &mut TaskBuilder) {
+    let mut tags = Vec::new();
+    for pair in pairs {
+        match pair.as_rule() {
+            Rule::task_name => {
+                task_builder.name(pair.into_inner().as_str().to_string());
+            }
+            Rule::task_tag => {
+                tags.push(Tag::from(pair.into_inner().as_str()));
+            }
+            Rule::task_plugin => {
+                let mut plugin_builder = PluginBuilder::new();
+                create_plugin(pair.into_inner(), &mut plugin_builder);
+                task_builder.plugin(plugin_builder.build());
+            }
+            r => assert!(false, "not a task part {:?}", r),
+        };
+    }
+    task_builder.tags(tags);
+}
+
+fn create_plugin(pairs: Pairs<Rule>, plugin_builder: &mut PluginBuilder) {
+    let mut options = Vec::new();
+    for pair in pairs {
+        match pair.as_rule() {
+            Rule::ident => {
+                plugin_builder.name(Identifier::new(pair.as_str()));
+            }
+            Rule::plugin_option => {
+                options.push(create_option(pair.into_inner()));
+            }
+            r => assert!(false, "not a plugin part {:?}", r),

```

Comment:
I would prefer to not have code paths in the parser that can panic. Rather, we should try to provide helpful errors to the user. 

Specifically, I really like the output of the [`miette`](https://docs.rs/miette/latest/miette/) crate

Reply:
using miette now